src/useCookie.ts

```diff
@@ -0,0 +1,125 @@
+import { useState, useEffect } from 'react'
+
+import noop from './shared/noop'
+import isClient from './shared/isClient'
+import isDevelopment from './shared/isDevelopment'
+import isAPISupported from './shared/isAPISupported'
+import createHandlerSetter from './factory/createHandlerSetter'
+
+import { BRHGeolocationPositionError } from './shared/types'
+
+export enum ECookieSameSite {
+  STRICT = 'strict',
+  LAX = 'lax',
+  NONE = 'none',
+}
+
+interface ICookieStoreDeleteOptions {
+  name?: string;
+  domain?: string;
+  path?: string;
+}
+
+interface ICookieInit extends ICookieStoreDeleteOptions {
+  sameSite?: ECookieSameSite;
+}
+
+interface ICookieInitWithNameAndValue extends ICookieInit {
+  name?: string;
+  value?: string;
+}
+
+export interface IOptions extends ICookieInit {
+  defaultValue?: string;
+}
+
+interface ICookieStore {
+  get: (key: string) => Promise<ICookieInitWithNameAndValue>;
+  set: (options: ICookieInitWithNameAndValue) => Promise<void>;
+  delete: (options: ICookieStoreDeleteOptions) => Promise<void>;
+}
+
+const useCookie = (key: string, options?: IOptions) => {
+  if (!isClient) {
+    if (!isDevelopment) {
+      // eslint-disable-next-line no-console
+      return console.warn(
+        'Please be aware that cookieStore could not be available during SSR',
+      )
+    }
+
+    return Object.freeze({
```

Comment:
that's a cool hack to avoid breaking the consistency of response, but you can store this into a constant outside of the hook's scope to avoid it being re-defined every render

```
const hookNotSupportedResponse = Object.freeze({
      onError: noop,
      updateCookie: noop,
      deleteCookie: noop,
      cookieValue: options?.defaultValue,
    })

const useCookie = (...)  => {
    if(!isClient) {
        .... 
        return hookNotSupportedResponse;
    }
   
}

```